bundler/lib/bundler/templates/Executable.bundler

```diff
@@ -47,7 +47,7 @@ m = Module.new do
   def lockfile
     lockfile =
       case File.basename(gemfile)
-      when "gems.rb" then gemfile.sub(/\.rb$/, gemfile)
+      when "gems.rb" then gemfile.sub(/\.rb$/, ".locked")
```

Comment:
After the change, the tests all passed. But I do not really know how write a test for this. Would love to have maintainer's help on this.

Reply:
:thinking: This was added probably here https://github.com/rubygems/rubygems/commit/cdcda68e79ef3d6b5df78ce41066f15d7703395e#diff-fc635269f10cd2189b47bf5ae555ee4bf14882724b5871383904cd8a07de69ddR34. :older_adult: 

Ad testing, have you checked https://github.com/rubygems/rubygems/blob/master/bundler/spec/runtime/executable_spec.rb

Reply:
👋 @simi , thanks for the tip! I have added some tests which are failing on `master` and green on this branch!